docs/source/links/state.md

```diff
@@ -0,0 +1,609 @@
+   the cache.
+2. The resolver should return an object with a `__typename` property unless
+   you've overridden the `dataIdFromObject` function to not use `__typename` for
+   cache keys. This is necessary for Apollo Client to [normalize the data in the
+   cache](/docs/react/basics/caching.html#normalization) properly.
+3. Resolver functions can return a promise if you need to perform asynchronous
+   side effects.
+
+If any of that sounds confusing, I promise it will be cleared up by the end of
+this section. Keep on reading! 😀
+
+<h3 id="default">Default resolvers</h3>
+
+You don't have to specify resolver functions for every field, however. If the
+return value from the parent object has the same property names as the fields
+requested in the child object, you won't need to specify a resolver. This is
+called a [default resolver](docs/graphql-tools/resolvers.html#Default-resolver).
+
+```js
+const getUser = gql`
+  query {
+    user(id: 1) @client {
+      name {
+        last
+        first
+      }
+    }
+  }
+`;
+```
+
+For this query, you will need to specify a resolver for `Query.user` in your
+resolver map. If `Query.user` returns an object with a name property that
+corresponds to an object with last and first properties, you do not need to
+specify any additional resolvers. GraphQL takes care of that for you!
+
+<h3 id="resolver-signature">Resolver signature</h3>
+
+The signature of a resolver function is the exact same as resolver functions on
+the server built with `graphql-tools`. Let's quickly recap the four parameters
+of a resolver function:
+
+```js
+fieldName: (obj, args, context, info) => result;
+```
+
+1. `obj`: The object containing the result returned from the resolver on the
+   parent field or the `ROOT_QUERY` object in the case of a top-level query or
+   mutation. Don't worry about this one too much for `apollo-link-state`.
+2. `args`: An object containing all of the arguments passed into the field. For
+   example, if you called the field with `networkStatus(isConnected: true)`, the
+   `args` object would be `{ isConnected: true }`.
+3. `context`: The context object, which is shared by all links in the Apollo
+   Link chain. The most important thing to note here is that we've added the
+   Apollo cache to the context for you, so you can manipulate the cache with
+   `cache.writeData({})`. If you want to set additional values on the context,
+   you can set them from [within your
+   component](docs/react/basics/queries.html#graphql-config-options-context) or
+   by [using `apollo-link-context`](/docs/link/links/context.html).
+4. `info`: Information about the execution state of the query. You will probably
+   never have to use this one.
+
+For further exploration, check out the [`graphql-tools`
+docs](/docs/graphql-tools/resolvers.html#Resolver-function-signature).
+
+<h3 id="async">Async resolvers</h3>
+
+`apollo-link-state` supports asynchronous resolver functions. This can be useful
```

Comment:
Somewhere here maybe be explicit that these functions can either be `async` functions or ordinary functions that return a Promise.